File: TelerikUIExamples/TelerikUIExamples/CustomAnnotation.m
//
//  CustomAnnotation.m
//  TelerikUIExamples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

#import "CustomAnnotation.h"
#import <TelerikUI/TelerikUI.h>
#import "MyAnnotation.h"

@implementation CustomAnnotation
{
    TKChart *_chart;
}

- (instancetype)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    _chart = [[TKChart alloc] initWithFrame:self.exampleBoundsWithInset];
    _chart.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    [self.view addSubview:_chart];
    
    NSArray *months = @[@"Jan", @"Feb", @"Mar", @"Apr", @"May", @"Jun"];
    NSArray *values = @[@95, @40, @55, @30, @76, @34];
    NSMutableArray *array = [NSMutableArray new];
    for (int i = 0; i<months.count; i++) {
        [array addObject:[[TKChartDataPoint alloc] initWithX:months[i] Y:values[i]]];
    }
    TKChartAreaSeries *series = [[TKChartAreaSeries alloc] initWithItems:array];
    series.style.pointShape = [[TKPredefinedShape alloc] initWithType:TKShapeTypeCircle andSize:CGSizeMake(10, 10)];
    [_chart addSeries:series];
    
    // Add my custom annotation
    TKPredefinedShape *shape = [[TKPredefinedShape alloc] initWithType:TKShapeTypeStar andSize:CGSizeMake(20, 20)];
    MyAnnotation *a = [[MyAnnotation alloc] initWithShape:shape X:@"Mar" Y:@60 forSeries:series];
    a.fill = [TKSolidFill solidFillWithColor:[UIColor blueColor]];
    a.stroke = [TKStroke strokeWithColor:[UIColor yellowColor] width:3];
    [_chart addAnnotation:a];
    
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/CustomAnnotation.swift
//
//  CustomAnnotation.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class CustomAnnotation: ExampleViewController {
    
    let chart = TKChart()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        chart.frame = self.exampleBoundsWithInset
        chart.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.view.addSubview(chart)
        
        let months = ["Jan", "Feb", "Mar", "Apr", "May", "Jun"]
        let values = [95, 40, 55, 30, 76, 34]
        var array = [TKChartDataPoint]()
        for i in 0..<months.count {
            array.append(TKChartDataPoint(x: months[i], y: values[i]))
        }
        let series = TKChartAreaSeries(items: array)
        series.style.pointShape = TKPredefinedShape(type: TKShapeType.Circle, andSize: CGSizeMake(10, 10))
        chart.addSeries(series)
        
        let shape = TKPredefinedShape(type: TKShapeType.Star, andSize: CGSizeMake(20, 20))
        let a = MyAnnotation(Shape: shape, X: "Mar", Y: 60, forSeries: series)
        a.fill = TKSolidFill(color: UIColor.blueColor())
        a.stroke = TKStroke(color: UIColor.yellowColor(), width: 3)
        chart.addAnnotation(a)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
}
